### DISCUSSION TITLE
Use Orchard Core in Blazor web app

### DISCUSSION THREAD

**Discussion by agriffard** [OrchardCMS/OrchardCore#14722] [Ideas]
I am trying to reproduce in .NET 8 with a new Blazor web app what I have been able to do in .NET 6 and that I described here https://github.com/OrchardCMS/OrchardCore/discussions/13611:

Make OC and Blazor work together and use the @OrchardHelper in a decoupled mode.

Unfortunately, I don't manage to make it work. 
The first step would be to display the OC setup screen but it does not appear.

If someone has a suggestion to go further, please tell me.

**Comment by agriffard** [comment]
I created a sample Blazor web app: https://github.com/agriffard/Zeugma

I use the preview 1.8.* version of OC packages.

**Comment by ns8482e** [comment]
No need to call `AddServerSideBlazor` explicitly. as that is now being called from `AddInteractiveServerComponents` .

In `.ConfigureSevices()` replace  `AddServerSideBlazor` with following
```cs
 svc.AddRazorComponents()
	 .AddInteractiveServerComponents()
	 .AddInteractiveWebAssemblyComponents();
```
and moving statements after `UseOrchardCore` into `Configure( (a,b,c)=>)` and replace `MapBlazorHub` with following
```cs
        a.UseAntiforgery();
	b.MapRazorComponents<App>()
		.AddInteractiveServerRenderMode()
		.AddInteractiveWebAssemblyRenderMode()
		.AddAdditionalAssemblies(typeof(Counter).Assembly);
	
```

**Comment by agriffard** [comment]
Thank you @ns8482e, I made the change you suggested in Program.cs but it is still not working, the OC setup does not appear.

https://github.com/agriffard/Zeugma/blob/main/src/Zeugma/Program.cs

**Reply by ns8482e** [reply]
Setup doesn't appear because `Home.razor` is taking over the root `/`.

change route to `@page /home` for `Home.razor`

**Reply by agriffard** [reply]
Thx @ns8482e I managed to launch a setup with your tip.

I added a Recipe and a Page.razor with `@page "/p/{alias}"` that allows you to get the data from a Page content item with the corresponding alias.

**Comment by agriffard** [comment]
![Animation](https://github.com/OrchardCMS/OrchardCore/assets/703248/5519c334-d805-4af0-8ee5-592328a16be6)

**Reply by ns8482e** [reply]
@agriffard  additionally in `App,.razor` you can configure `<base href="/@ShellScope.Context.Settings.RequestUrlPrefix/" />`  to run blazor pages for different tenants.

**Comment by psijkof** [comment]
With the help of the PR that is open for Blazor and OC and this discussion, I came up with this so far, enabling multitenancy.

https://github.com/psijkof/oc-mt-blazor8-iserver/tree/master

**Reply by hishamco** [reply]
Can you demo it on the next meeting?

**Reply by psijkof** [reply]
I am happy to walk through it. When is the next meeting?

**Reply by hishamco** [reply]
On Tuesday, you can check the Podcast link in https://orchardcore.net/ 

it's 7:00 PM GTM nowadays

**Reply by psijkof** [reply]
I did some refactoring and now also show how to get the base url from httpcontext, sitesettings and ultimately the shellscope requesturlprefix.

I will try soon to add some more text to the PR. Thank you @hishamco

**Reply by hishamco** [reply]
Thanks for your demo Sijmen
